src/db/nedb.js

```diff
@@ -63,6 +64,29 @@ class DBHelper {
     }
   }
 
+  /*
+  * Returns the fields of the object in one of the tables searched by the query.
+  * @param db The DB table.
+  * @param query {Object} The query by items.
+  * @param fields {Array} The fields to return for the found item in an array.
+  */
+  static async findOne(db, query, fields) {
+    let fieldsObj;
+    if (!_.isEmpty(fields)) {
+      fieldsObj = {};
+      _.each(fields, (field) => {
```

Comment:
we might have to have this on multiple lines so it's not returning each time, but maybe test out to see if we can do:
`_.each(fields, (field) => fieldsObj[field] = 1)`

also, can't we use reduce for this?
`const fieldsObj = (fields || []).reduce((obj, field) => obj[field] = 1, {});`

might have to be:
`const fieldsObj = (fields || []).reduce((obj, field) => { obj[field] = 1 }, {});`

Reply:
reduce does not work for this. shortened it without the brackets